lms/static/js/index_discovery/models/index_course_discovery.js

```diff
@@ -31,6 +31,20 @@
 
                 _(programs).each(function(program, idx) {
                     program.courses = _.pluck(program.courses, 'data');
+                    // TODO: Figure out why IndexCourseCard model does not handle properties absence.
+                    _(program.courses).each(function(course, idx) {
+                        if (!course.product) {
+                            course.product = false;
+                        }
+
+                        if (!course.downloadable) {
+                            course.downloadable = false;
+                        }
+
+                        if (!course.external_product_link) {
+                            course.external_product_link = '';
+                        }
+                    });
```

Comment:
`if (course.product === undefined)`?

Reply:
or product can only be true?

Reply:
@sendr it can be `false`. But in either case (`false` or `undefined`) it should be `false` so the code is universal.